mono/metadata/handle.h

```diff
@@ -401,19 +401,15 @@ This is why we evaluate index and value before any call to MONO_HANDLE_RAW or ot
 #define MONO_HANDLE_ARRAY_SETREF(HANDLE, IDX, VALUE) do {	\
 		int __idx = (IDX);	\
    		MonoObjectHandle __val = MONO_HANDLE_CAST (MonoObject, VALUE);		\
-		do {							\
-			MONO_HANDLE_SUPPRESS_SCOPE(1);			\
-			mono_array_setref_fast (MONO_HANDLE_RAW (MONO_HANDLE_UNSUPPRESS (HANDLE)), __idx, MONO_HANDLE_RAW (__val)); \
-		} while (0);						\
+		MONO_HANDLE_SUPPRESS_SCOPE(1);			\
+		mono_array_setref_fast (MONO_HANDLE_RAW (MONO_HANDLE_UNSUPPRESS (HANDLE)), __idx, MONO_HANDLE_RAW (__val)); \
 	} while (0)
 
 #define MONO_HANDLE_ARRAY_SETVAL(HANDLE, TYPE, IDX, VALUE) do {		\
 		int __idx = (IDX);					\
    		TYPE __val = (VALUE);					\
-		do {							\
-			MONO_HANDLE_SUPPRESS_SCOPE(1);			\
-			mono_array_set (MONO_HANDLE_RAW (MONO_HANDLE_UNSUPPRESS (HANDLE)), TYPE, __idx, __val); \
-		} while (0);						\
+		MONO_HANDLE_SUPPRESS_SCOPE(1);				\
```

Comment:
same as setref - either `MONO_HANDLE_SUPPRESS_SCOPE(1)` as first thing inside braces, or else just wrap the `mono_array_set` in a a `MONO_HANDLE_SUPPRESS`